enclave/src/polkadex_gateway.rs

```diff
@@ -350,4 +442,358 @@ pub fn authenticate_user(main_acc: AccountId, proxy_acc: Option<AccountId>) -> R
 //         return Err(Default::default());
 //     }
 //     Ok(())
-// }
\ No newline at end of file
+// }
+
+pub fn settle_trade(trade: TradeEvent) -> Result<(), GatewayError> {
+    // Check if both orders exist and get them
+    let mut maker =
```

Comment:
It will work this way, but it would be more efficient to just lock the storage once and the load both orders and the remove lock instead of locking the same storage twice

Reply:
I will create a new issue for optimization stuff.

Reply:
I am sorry ... I didn't get it.

Reply:
I created an issue for that @haerdib ... I will do it later.